icu4c/source/tools/icuexportdata/icuexportdata.cpp

```diff
@@ -532,12 +532,17 @@ void computeDecompositions(const char* basename, const USet* backwardCombiningSt
     IcuToolErrorCode status("icuexportdata: computeDecompositions");
```

Comment:
Suggestion (optional):

1. Change this function to take `nonRecursive32` and `nonRecursiveBuilder` as arguments
2. Create a new function `computeDecompositionsAndWriteNonRecursive` and move the file-writing machinery to it
3. At the call site, change `computeDecompositions("nfd", ...)` to `computeDecompositionsAndWriteNonRecursive("nfd")`